src/MemoryTracing/MemoryTracingUtils.cpp

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "MemoryTracingUtils.h"
+
+#include <absl/strings/numbers.h>
+#include <absl/strings/str_split.h>
+#include <stdlib.h>
+
+#include <string>
+
+#include "GrpcProtos/Constants.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Profiling.h"
+#include "OrbitBase/ReadFileToString.h"
+#include "OrbitBase/Result.h"
+
+namespace orbit_memory_tracing {
+
+using orbit_grpc_protos::kMissingInfo;
+using orbit_grpc_protos::SystemMemoryUsage;
+
+std::optional<SystemMemoryUsage> ParseMemInfo(const std::string& meminfo_content) {
+  if (meminfo_content.empty()) return std::nullopt;
+
+  std::vector<std::string> lines = absl::StrSplit(meminfo_content, '\n');
+  if (lines.empty()) return std::nullopt;
+
+  SystemMemoryUsage memory_info;
+  memory_info.set_total_kb(kMissingInfo);
+  memory_info.set_free_kb(kMissingInfo);
+  memory_info.set_available_kb(kMissingInfo);
+  memory_info.set_buffers_kb(kMissingInfo);
+  memory_info.set_cached_kb(kMissingInfo);
+
+  constexpr size_t kNumLines = 5;
+  size_t end = lines.size() > kNumLines ? kNumLines : lines.size();
+  int mem_total, mem_free, mem_available, mem_buffers, mem_cached;
+  for (size_t i = 0; i < end; ++i) {
+    const std::string& current_line = lines[i];
+    std::vector<std::string> splits = absl::StrSplit(current_line, ' ', absl::SkipWhitespace{});
```

Comment:
maybe use absl::Span<> here and split in 2 by using absl::MaxSplits(' ', 2)?

Reply:
Thanks for pointing out this and providing helpful suggestions! 

I'm considering that lines in `/proc/meminfo` end with "KB", for example "MemTotal:       1234567 kB". If we use `absl::MaxSplits(' ', 2)`, then the "KB" will be included in the second element of the splitting result.

Reply:
hmm, does it mean we need to parse KB MB GB as well? they seem to be ignored now.

Reply:
Yes. In the current version I split  "MemTotal: 1234567 kB" into three pieces, i.e., "MemTotal:", "1234567" and "kB", and only extract information from the second splitting piece. 
With `absl::MaxSplits(' ', 2)` we will get two pieces, i.e.,  "MemTotal:" and "1234567 kB". Do we need also parse KB, MB, GB? I'm assuming that the content of `/proc/meminfo` has a fixed format.